src/name-extractor/__tests__/autocomplete.ts

```diff
@@ -1,547 +1,524 @@
-      if (baz) {\n\
-        let bar = 1;\n\
+      function foo2(bar2) {\n\
+        b\n\
       }\n\
-      b\n\
-    }\
-  '
-  const line = 5
-  const col = 2
-  const [extractedNames] = await getNames(code, line, col, createContext(0))
-  const expectedNames: NameDeclaration[] = [
-    { name: 'foo', meta: 'func', score: 0 },
-    { name: 'baz', meta: 'param', score: 1 }
-  ]
-  expect(new Set(extractedNames)).toMatchObject(new Set(expectedNames))
-})
-
-// Blocks
-
-test('Test that declaration in blocks cannot be accessed outside block', async () => {
-  const code: string =
-    '\
-    {\n\
-      let foo = 1;\n\
-    }\n\
-    f\
-  '
-  const line = 4
-  const col = 1
-  const expectedNames: NameDeclaration[] = []
-  const [extractedNames] = await getNames(code, line, col, createContext(0))
-  expect(new Set(extractedNames)).toMatchObject(new Set(expectedNames))
-})
-
-test('Test that declaration outside blocks can be accessed inside block', async () => {
-  const code: string =
-    '\
-    let bar = 1;\n\
-    {\n\
-      let baz = 1;\n\
-      b\n\
-    }\n\
-  '
-  const line = 4
-  const col = 2
-  const expectedNames: NameDeclaration[] = [
-    { name: 'baz', meta: 'let', score: 1 },
-    { name: 'bar', meta: 'let', score: 0 }
-  ]
-  const [extractedNames] = await getNames(code, line, col, createContext(0))
-  expect(new Set(extractedNames)).toMatchObject(new Set(expectedNames))
-})
-
-// Anonymous functions
-
-test('Test that declaration outside of anonymous functions can be accessed inside', async () => {
-  const code: string =
-    '\
-    let foo = () => { \n\
-      let baz = 1;\n\
-      b\n\
-    }\n\
-    let bar = 3;\n\
-  '
-  const line = 4
-  const col = 1
-  const expectedNames: NameDeclaration[] = [
-    { name: 'bar', meta: 'let', score: 1 },
-    { name: 'foo', meta: 'let', score: 0 },
-    { name: 'baz', meta: 'let', score: 2 }
-  ]
-  const [extractedNames] = await getNames(code, line, col, createContext(0))
-  expect(new Set(extractedNames)).toMatchObject(new Set(expectedNames))
-})
-
-test('Test that declaration inside anonymous functions can be accessed in body', async () => {
-  const code: string =
+    ',
+    2,
+    3,
+    [
+      { name: 'foo2', meta: DeclarationKind.KIND_FUNCTION, score: 1 },
+      { name: 'foo1', meta: DeclarationKind.KIND_FUNCTION, score: 0 },
+      { name: 'bar1', meta: DeclarationKind.KIND_PARAM, score: 2 },
+      { name: 'baz1', meta: DeclarationKind.KIND_PARAM, score: 3 }
+    ],
+    [{ name: 'baz2', meta: DeclarationKind.KIND_CONST }]
+  ],
+  // For loops
+  [
+    'Test accessing local block in for-loop parameter',
+    `
+    let bar = 1;
+    let baz = 2;
+    for (b) {
+    `,
+    3,
+    6,
+    [
+      { name: 'baz', meta: DeclarationKind.KIND_LET, score: 1 },
+      { name: 'bar', meta: DeclarationKind.KIND_LET, score: 0 }
+    ]
+  ],
+  [
+    'Test accessing for-loop parameter in for-loop body',
     '\
+      for (let foo=10;) {\n\
+        f\n\
+      }\
```

Comment:
Can we replace this (and all the other occurrences) with a template literal instead?